src/encoded/static/components/experiment_series.js

```diff
@@ -192,21 +384,82 @@ class ExperimentSeriesComponent extends React.Component {
-        if (viewableDatasets.length > 0) {
+        const allRows = [];
+        const viewableDatasets = this.state.viewableDatasets;
+        if (Object.keys(viewableDatasets).length > 0) {
+            // Add the "Add all to cart" button and internal tags from all related datasets.
             addAllToCartControl = (
                 <div className="experiment-table__header">
                     <h4 className="experiment-table__title">{`Experiments in experiment series ${context.accession}`}</h4>
-                    <CartAddAllElements elements={viewableDatasets} />
+                    <CartAddAllElements elements={Object.values(viewableDatasets)} />
                 </div>
             );
 
             // Collect unique internal_tags from all relevant experiments.
-            internalTags = _.uniq(viewableDatasets.reduce((allInternalTags, experiment) => (
+            internalTags = _.uniq(Object.values(viewableDatasets).reduce((allInternalTags, experiment) => (
                 experiment.internal_tags && experiment.internal_tags.length > 0 ? allInternalTags.concat(experiment.internal_tags) : allInternalTags
             ), []));
+
+            // Experiment table data
+            Object.values(viewableDatasets).forEach((experiment) => {
+                // Select one analysis based on file assembly and file count
+                // for extracting quality metrics
+                const filesByDesiredAssembly = experiment.files.reduce(
+                    (fileAtIds, f) => {
+                        if (selectedAssembly.includes(f.assembly)) {
+                            fileAtIds.push(f['@id']);
+                        }
+                        return fileAtIds;
+                    },
+                    []
+                );
+                const analyses = experiment.analyses || [];
+                const selectedAnalysis = analyses.filter(
+                    analysis => analysis.files.every(f => filesByDesiredAssembly.includes(f))
+                ).sort((a, b) => a.length - b.length)[0] || { files: [] };
+                experiment.files = experiment.files.filter(
+                    f => selectedAnalysis.files.includes(f['@id'])
+                );
+                // Number of subrows to be shown corresponds to the number of biological replicates. Show at least one by default.
+                let bioRepNums = _.uniq(
+                    experiment.files.map(f => f.biological_replicates).flat()
+                );
+                if (bioRepNums.length === 0) {
+                    bioRepNums = [''];
+                }
+                // Extract experiment metadata
+                const rowCount = bioRepNums.length;
+                const datasetRows = [];
+                bioRepNums.sort((a, b) => a - b).forEach((num, i) => {
+                    const cellMeta = { bioRepNum: num, isAuthorized, rowCount };
+                    datasetRows.push(
+                        <tr key={experiment['@id'].concat('_', i)}>
```

Comment:
I think `num` will be unique for each row, right? It would be better to use `num` instead of `i` because `i` always starts at 0 even if you change the order of the rows, and React might not notice you changed the order. If you use `num` instead, then React will notice if you change the order because `num` will move with the data.